search-improvements.rst

```diff
@@ -0,0 +1,143 @@
+`Search as You Type and Autocomplete`_, `Code Search`_ and many more.  All of his search related work can be
+looked in the `Search Project Board`_.
+
+
+Sphinx/MkDocs Built in Search vs Read The Docs Search
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Sphinx/MkDocs already have built in search functionality. But the features are very limited and heavy.
+At Read The Docs, we have felt the limitations and therefore we index the documentations in our
+Elasticsearch_ index so that we can provide better results to the users. Currently, we do not index
+MkDocs documents to Elasticsearch_, but `any kind of help is welcome`_.
+
+New Features
+^^^^^^^^^^^^
+The number of features that can be included in Search is infinity. There are always way to improve.
+In the 3 months of full time work, our GSoC student have implemented couple of features including
+many bug fixes. Some of the major features are as following:
+
+- `Exact Matching Search`_: Its a highly needed feature for Read The Docs. Now you can search for
+  exact word in documentations by having your query inside quotation (`""`). So if you search
+  for `"Here is foo"` (with the quotation), you will get all the documentations where the full
+  `Here is foo` phrase exist.
+
+- Simple Query String Syntax support: Now you can use `Simple Query String Syntax`_ for
+  searching. For example, if you search with `Mozilla +-Firefox`, you will get documentations
+  where the `Mozilla` word is present, but `Firefox` word is not present.
+  For more information, you can look at `Simple Query String Syntax`_ documentation.
+
+- `Case Insensitive Search`_: The search is now case insensitive. So if you search for `Foo`,
+  you will get all the documentations which have one of either `Foo`, `FOO`, `foo` or `fOO`.
+
+- `Improved Result order`_: The result order is improved dramatically. Therefore if you Search
+  with `Foo Bar`, you will first see the documentations which have both `Foo Bar`, then
+  you will see other documentations which have either `Foo` or `Bar`
+
+- `Auto Removing from Search Index`_: In the past, if any page got removed from documentations,
+  the page was still available in documentations. But from now on, the page will be removed
+  automatically from the search index as soon as its no longer in the documentation.
+
+- `Zero Downtime Indexing`_: As search is a important part of documentation, there will be no
+  downtime while we reindex our Elasticsearch Index. So the search will be much more reliable
+  than before.
+
+
+Performance Improvements
+^^^^^^^^^^^^^^^^^^^^^^^^
+Performance is always improtant for search. Because of our search was fast enough,
+we did not focus to improve the performance. As we have rewritten all the codes with
+new search backend engine, we got 4x improved performance as well.
+
+There are 4 kind of search functionality in Read The Docs. The improvements are as following:
+
+- `File Search`_
```

Comment:
I'd give a one sentence blurb explaining what each of "File Search", "Projects Search", are. Those two are somewhat obvious but "Doc Search" and "Project Doc Search" are not.